refactor(parser): Remove Lexer peeking for modifiers

leaysgur opened the pull request:
Part of #11194 

NOTE: In TypeScript parser.ts, `next_token_can_follow_modifier` seems to be the only place to handle these checks.

https://github.com/microsoft/TypeScript/blob/b504a1eed45e35b5f54694a1e0a09f35d0a5663c/src/compiler/parser.ts#L2765

But we have two places, may revisit it later.

Boshen commented:
### Merge activity

* **May 22, 6:05 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
refactor(parser): Remove Lexer peeking for modifiers (#11228)

Part of #11194

NOTE: In TypeScript parser.ts, `next_token_can_follow_modifier` seems to be the only place to handle these checks.

https://github.com/microsoft/TypeScript/blob/b504a1eed45e35b5f54694a1e0a09f35d0a5663c/src/compiler/parser.ts#L2765

But we have two places, may revisit it later.
